docs/3_NEW_TEMPLATE.md

```diff
@@ -1,15 +1,406 @@
+    | azurerm_function_app | According to the Terraform docs, this is the only resource unique to an Azure Function. This resource block will be declared within the module. |
+    | azurerm_app_service_plan | The azure function app needs to live within an app service plan so we have to instantiate the existing module that encapsulates the app service plan. |
+    | azurerm_storage_account | The azure function app's ephemeral state needs a dedicated storage account, however, after further inspection, the existing module for the storage account will not be usuable as it does not support a connection string as an output due to security best practices. We'll have to declare the azurerm_storage_account resource directly from the template and avoid using it's module.  |
+    | azurerm_resource_group | Most Azure infrastructure lives in a resource group container so this will be a part of the configuration as well.  |
+
+1. **Define inputs** - When a CIT instantiates a module, it will configure that module using it's exposed input variable names. These variables will pass values to the attributes of the resource blocks internal to the module. These inputs have also been defined for you below:
+
+    | *azurerm_function_app* resource attribute | Scope | Required | Input Variable Name | Description |
+    |--------|-------------|-------------|-----------|-----------|
+    | name | Input | yes | `azure_function_name` | A name for the function app and how it will be identified within your Azure subscription. |
+    | name | Input | yes | `azure_function_name_prefix` | A prefix name for appending unique values to the azure function name. |
+    | resource_group_name | Input | yes | `resource_group` | Most Azure infrastructure lives in a resource group container of your choice. By making this an input, each module instance can have a different resouce group. |
+    | location | Input | yes | `resource_group_location` | The geo-location here should derive from the geo-location that the resource group name lives in. |
+    | app_service_plan_id  | Input | yes | `app_service_plan_id` | This input implies that the azure function resource will live within an app service plan. |
+    | storage_connection_string | Input | yes | `storage_connection_string` | This is the storage account in which the ephemeral state for an Azure Function will be orchestrated when the endpoint is invoked. |
+    | app_settings | Internal | no | `-` | { environment = "hw-from-scratch" } - We will provide a hard-coded key-value pair as an example that does not require an input. Value will not be passed from a CIT. |
+
+1. **Define outputs** - A module instance will only output values that it's been pre-configured to output. It's **best practice** to configure module instance outputs so that you can validate expected results. These results are visible in standard out if passed to the template when running the terraform plan and apply steps. These outputs are defined for you below:
```

Comment:
```suggestion
1. **Define outputs** - A module instance will only output values that it's been pre-configured to output. It's **best practice** to configure module outputs because it enables module composition. These results are visible in standard out if passed to the template when running the terraform plan and apply steps. These outputs are defined for you below:
```